libp2p/peer/peerdata_interface.py

```diff
@@ -1,54 +1,61 @@
 from abc import ABC, abstractmethod
+from typing import (
+    Any,
+    List,
+    Sequence,
+)
+
+from multiaddr import Multiaddr
 
 
 class IPeerData(ABC):
 
     @abstractmethod
-    def get_protocols(self):
+    def get_protocols(self) -> List[str]:
         """
         :return: all protocols associated with given peer
         """
 
     @abstractmethod
-    def add_protocols(self, protocols):
+    def add_protocols(self, protocols: Sequence[str]) -> None:
         """
         :param protocols: protocols to add
         """
 
     @abstractmethod
-    def set_protocols(self, protocols):
+    def set_protocols(self, protocols: Sequence[str]) -> None:
         """
         :param protocols: protocols to add
         """
 
     @abstractmethod
-    def add_addrs(self, addrs):
+    def add_addrs(self, addrs: Sequence[Multiaddr]) -> None:
         """
         :param addrs: multiaddresses to add
         """
 
     @abstractmethod
-    def get_addrs(self):
+    def get_addrs(self) -> List[Multiaddr]:
         """
         :return: all multiaddresses
         """
 
     @abstractmethod
-    def clear_addrs(self):
+    def clear_addrs(self) -> None:
         """
         Clear all addresses
         """
 
     @abstractmethod
-    def put_metadata(self, key, val):
+    def put_metadata(self, key: Any, val: Any) -> None:
         """
         :param key: key in KV pair
         :param val: val to associate with key
         :raise Exception: unsuccesful put
         """
 
     @abstractmethod
-    def get_metadata(self, key):
+    def get_metadata(self, key: Any) -> Any:
```

Comment:
I failed to find `PeerData` in go-libp2p. But I guess the key here is `str` from https://github.com/libp2p/go-libp2p-core/blob/master/peerstore/peerstore.go#L80-L86
```suggestion
    def get_metadata(self, key: str) -> IPeerMetadata:
```